pages/en/lb4/Decorator.md

```diff
@@ -0,0 +1,125 @@
+
+Some components also exports decorators for you:
+- [authentication](#authentication-decorators)
+- [repository](#repository-decorators)
+
+*Janny: Most examples are documented in a certain concept that a decorator works closely with, I am not sure should I turn this page to something like a reference:*
+
+- some usage are well documented, I linked to certain section
+- some usage can be found in unit test, while integration example is missing, should I create on this page?
+- some have integration test, modify it and add on this page?
+- Do I need to document what metadata each decorator adds to the reflector and document in the reference?
+- I am confused with some decorator can be used by either @mydecorator or @inject(‘mydecorator’), are they equal? 
+
+## Route Decorators
+- @get, @post, @put, @patch, @del
+
+  Register route for a controller by rest decorator, for their usage, refer to [Routing to Controllers](controller.htm#Routing-to-Controllers)
+
+- @api
+
+  Register route for a controller by api decorator, more details and example refer to [Specifying Controller APIs](controllers.htm#Specifying Controller APIs)
+
+- @operation
+
+  Expose a Controller method as a REST API operation. Usage refer to [code](https://github.com/strongloop/loopback-next/blob/debe221ee7eb07b6d6962416b79592c8f13d1e91/packages/core/src/router/metadata.ts#L230-L234)
+
+  *Janny: Need example in doc?*
+
+- @param
+
+  Describe an input parameter of a Controller method. An example can be found in [Writing Controller methods](controller.htm#Writing-Controller-methods), more usage refer to [tsDoc](https://github.com/strongloop/loopback-next/blob/debe221ee7eb07b6d6962416b79592c8f13d1e91/packages/core/src/router/metadata.ts#L252-L272)
+
+  *Janny: Change link to tsDoc page when it's done*
+  
+## Dependency Injection
+
+@inject is a decorator to annotate method arguments for automatic injection by LoopBack IoC container. It can only be applied on a **method parameter** or a **property**. Basiclly it defines a reflector metadata for the target class.
+
+A quick example can be found in [test case](https://github.com/strongloop/loopback-next/blob/9769e85306a8a866d71f133b1d0c013dde0d43c2/packages/context/test/unit/inject.test.ts#L21-L28)
+
+It's been documented as a LoopBack core concept in [Dependency Injection](Dependency-Injection.htm)
+
+*Janny: we miss a list of all metadata keys we can inject*
+
+## Authentication Decorators
+
+- @authenticate
+
+Mark a controller method as requiring authenticated user, it takes in a strategy name. Example refer to [test case](https://github.com/strongloop/loopback-next/blob/9769e85306a8a866d71f133b1d0c013dde0d43c2/packages/authentication/test/acceptance/basic-auth.ts#L104)
```

Comment:
fix link hash to be master.

Reply:
Better yet, don't link users to tests, but to the documentation for those plugins. If they don't have any, that's fine.